db/collection.go

```diff
@@ -264,6 +286,27 @@ func (db *db) updateSchema(
 				},
 			}
 
+			for i, globalField := range schema.Fields {
+				isNew := true
+				for _, localField := range col.Fields {
+					if localField.Name == globalField.Name {
```

Comment:
thought: There will be limitations here if we allow a field name to be changed locally. The way things are currently saved, a safer way to do this would be to use the global field index and the local field ID.

Reply:
I can't see us ever wanting to change field names cant be changed locally to differ from schema field names (guess we could have a local `label` property, but that is not its name).

At the moment the field name is it's global primary identity if/when(?) that changes we can/should of course change the way everything references global field definitions.

Reply:
Ah I had missed a key part of the CollectionFieldDescription.Name doc. Thanks for clarifying.